src/praisonai-train/tests/unit/data/test_dedup.py

```
@@ -0,0 +1,222 @@
+    assert len(kept) == 1
+
+
+# ── Regression: reviewer-reported edge cases ──────────────────────────────────
+
+def test_shared_empty_index_is_reused_across_batches():
+    # A freshly-created shared index is empty (falsy); it must still be reused so
+    # cross-batch near-dups are caught on the first call, not silently discarded.
+    shared = MinHashLSH(threshold=0.7, num_perm=128, seed=1)
+    b1 = _rows("translate this tamil line into fluent english please")
+    b2 = _rows("translate this tamil line into fluent english now")  # near-dup
+    k1, d1 = near_dedup(b1, {"near_dup_method": "minhash"}, index=shared)
```

Ruff (0.15.21)

[warning] Unpacked variable `k1` is never used

Prefix it with an underscore or any other dummy variable pattern

(RUF059)